### ISSUE TITLE
plugin/cache: README

### ISSUE THREAD

**Issue by miekg** [coredns/coredns#2248]
> A cache is divided into 256 shards, each holding up to 512 items by default - for a total size
of 256 * 512 = 131,072 items.

@fturib I can't remember (and not find it), where does the value of 512 come from?

**Comment by fturib** [comment]
It used to be the minimum size of a shard.
But was changed in this PR: https://github.com/coredns/coredns/pull/2065

the minimum is now 4

```
 ssize := size / shardSize
 if ssize < 4 {
    ssize = 4
 }
```

**Comment by chrisohaver** [comment]
Yeah - I missed that one.  Fixed in #2249.
Also, strictly speaking, our default cache is not 10000, due to shard size truncation, its 9984.

**Comment by miekg** [comment]
[ Quoting <notifications@github.com> in "Re: [coredns/coredns] plugin/cache:..." ]
>Yeah - I missed that one.  Fixed in #2249.
>Also, strictly speaking, our default cache is not 10000, due to shard size truncation, its 19*512 = 9728.

A thanks for the update. But the cache size is just diveded by the shardSize? We 
use 10,000 shardsize is 256, meaning per shard we only have 39 elements per map.

Where does the 512 come from in this calculation?

**Comment by chrisohaver** [comment]
I corrected my comment several hours ago. But you can’t see corrections in email....

> Yeah - I missed that one. Fixed in #2249.
Also, strictly speaking, our default cache is not 10000, due to shard size truncation, its 9984.

**Comment by miekg** [comment]
[ Quoting <notifications@github.com> in "Re: [coredns/coredns] plugin/cache:..." ]
>I corrected my comment several hours ago. But you can’t see corrections in email....

I think you have like 10s to change your comment, and then an email is sent. 
I've never seen an update being send out.

/Miek

--
Miek Gieben
